src/wallet/coincontrol.h

```diff
@@ -33,6 +33,10 @@ class PreselectedInput
     std::optional<int64_t> m_weight;
     //! The sequence number for this input
     std::optional<uint32_t> m_sequence;
+    //! The scriptSig for this input
+    std::optional<CScript> m_script_sig;
```

Comment:
in 53cb734591cabb37fc6e3d147a4715203cf15ddb

There is redundancy between `m_script_sig ` + `m_script_witness ` and `m_weight`. One can set all of them and reach and inconsistent state, which will lead to error.

Can we make the interface better?

Reply:
It is actually expected that both can be provided, and not match. We may be funding a transaction with other inputs that already have some signatures (e.g. a multisig where one of the parties is adding another input for fees, and one other party has already signed with sighash_single), so `m_script_sig` or `m_script_witness` may already exist, but are incomplete. `m_weight` may then also be provided, and it wouldn't match since it would account for the signatures that have not yet been added.